### PR TITLE
Sh/add request verification

### PR DESCRIPTION
**Pull request by dezuz**
### Description
Slack supports request verification mechanisms that are already used in a few projects so it's better to move the logic to this project and then use it wherever is needed. [Slack documentation](https://api.slack.com/authentication/verifying-requests-from-slack)

### Solution
To provide this logic I have done:
- added `slack-request-verifier` module
- added `SlackRequestVerifierFilter` class
- added unit tests to this class

### CODE REVIEW

**Comment by dezuz** [comment]
I found two possible ways to use this class in other projects:
- add a `bundle` and `feature` and register this filter for resources that are under the annotation
- extend this class, mark extended class with `@Provider` annotation and custom annotation that also would be used on the resource method, and bind it

File: slack-request-verifier/src/test/java/com.hubspot.slack.client.request.verifier/SlackRequestVerifierFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+import java.util.stream.Stream;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class SlackRequestVerifierFilterTest {
```
Comment (omotnyk) [inline slack-request-verifier/src/test/java/com.hubspot.slack.client.request.verifier/SlackRequestVerifierFilterTest.java:20]: 👍

File: slack-request-verifier/src/main/java/com/hubspot/slack/client/request/verifier/SlackRequestVerifierFilter.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  private static final String ENCODED_SIGNATURE_TEMPLATE = "%s=%s";
+  private static final String VERSION_NUMBER = "v0";
+
+  private final long requestExpirationTime;
+  private final String signingSecret;
+
+  public interface Factory {
+    SlackRequestVerifierFilter create(
+        @Assisted("requestExpirationTime") long requestExpirationTime,
+        @Assisted("signingSecret") String signingSecret
```
Comment (omotnyk) [inline slack-request-verifier/src/main/java/com/hubspot/slack/client/request/verifier/SlackRequestVerifierFilter.java:38]: IIRC, we don't have to specify `@Assisted` annotations on the factory params.

**Review by omotnyk — APPROVED** [review]
Thanks for working on this @dezuz. I left one small comment but overall it LGTM 👍 🚀

File: slack-request-verifier/src/main/java/com/hubspot/slack/client/request/verifier/SlackRequestVerifierFilter.java (hunk 2 above)
Comment (dezuz) [reply slack-request-verifier/src/main/java/com/hubspot/slack/client/request/verifier/SlackRequestVerifierFilter.java:38]: We do so in our internal projects so I thought that it works like that.
I will try to remove them

**Pull request merged**